### PR TITLE
Add emissions

### PR DESCRIPTION
**Pull request by mootz12**
* Add an emission feature that can be included on the wrapped or native token
* Fix event naming on votes contract

### CODE REVIEW

File: contracts/votes/src/balance.rs [hunk 1]
```diff
@@ -1,15 +1,119 @@
...
+/// Add tokens to an address's balance and update emissions and voting power
+///
+/// ### Arguments
+/// * `to` - The address to add the balance to
+/// * `amount` - The amount to add. This function does nothing if the amount is not greater than 0.
+///
+/// ### Panics
+/// If the total suppy exceeds the maximum value of the checkpoint
+pub fn mint_balance(e: &Env, to: &Address, amount: i128) {
+    if amount > 0 {
```
Comment (Ryang-21) [inline contracts/votes/src/balance.rs:21]: Is this check needed? You already check the amount in the mint call

File: contracts/votes/src/emissions.rs [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+            base_reserve: 10,
+            min_temp_entry_ttl: 100000,
+            min_persistent_entry_ttl: 100000,
+            max_entry_ttl: 100000,
+        });
+
+        let user = Address::generate(&e);
+        let contract = e.register_contract(None, crate::TokenVotes {});
+
+        let total_suppy = 1_123_456 * SCALAR_7;
```
Comment (Ryang-21) [inline contracts/votes/src/emissions.rs:385]: typo

File: contracts/votes/src/emissions.rs [hunk 3]
```diff
@@ -0,0 +1,866 @@
...
+            assert_eq!(result, 4 + 3095_4700000);
+            let balance = storage::get_balance(&e, &user);
+            assert_eq!(balance, 4 + 3095_4700000);
+            let votes = storage::get_voting_units(&e, &user);
+            assert_eq!(votes.to_checkpoint_data(), (123, 4 + 3095_4700000));
+        });
+    }
+
+    #[test]
+    fn test_claim_emissions_alread_updated_and_delegated() {
```
Comment (Ryang-21) [inline contracts/votes/src/emissions.rs:600]: typo

**Review by Ryang-21 — APPROVED** [review]
Found some typos but looks good

File: contracts/votes/src/balance.rs (hunk 1 above)
Comment (mootz12) [reply contracts/votes/src/balance.rs:21]: we likely don't need it, but we call mint balance in other places (like emissions), so having the check here seems reasonable

**Pull request merged**